compiler/code-gen/declarations.cpp

```diff
@@ -1141,16 +1141,26 @@ void ClassMembersDefinition::compile_msgpack_serialize(CodeGenerator &W, ClassPt
   //   packer.pack(field_1);
   //   ...
   //}
-  std::vector<std::string> body;
+  std::list<std::string> body;
   uint16_t cnt_fields = 0;
 
-  klass->members.for_each([&](ClassMemberInstanceField &field) {
-    if (field.serialization_tag != -1) {
-      auto func_name = fmt_format("vk::msgpack::packer_float32_decorator::pack_value{}", field.serialize_as_float32 ? "_float32" : "");
-      body.emplace_back(fmt_format("packer.pack({}); {}(packer, ${});", field.serialization_tag, func_name, field.var->name));
-      cnt_fields += 2;
+  ClassPtr the_klass = klass;
+
+  // Put class' fields along with all parent's fields.
+  while (the_klass) {
+    std::vector<std::string> body_inner;
+    the_klass->members.for_each([&](ClassMemberInstanceField &field) {
+      if (field.serialization_tag != -1) {
+        auto func_name = fmt_format("vk::msgpack::packer_float32_decorator::pack_value{}", field.serialize_as_float32 ? "_float32" : "");
+        body_inner.emplace_back(fmt_format("packer.pack({}); {}(packer, ${});", field.serialization_tag, func_name, field.var->name));
+        cnt_fields += 2;
+      }
+    });
+    for (auto it = body_inner.rbegin(); it != body_inner.rend(); ++it) {
+      body.push_front(*it);
```

Comment:
As you don't further need these values, it'd be nice to move them into `body`. E.g., `body.emplace_front(std::move(*it));`

Reply:
> As you don't further need these values, it'd be nice to move them into `body`. E.g., `body.emplace_front(std::move(*it));`

Surprisingly code logic got broken, compilation goes successful where should fail.